modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/storage/CurrencyMessageStorage.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.node.shared.infrastructure.snapshot.storage
+
+import cats.effect.std.MapRef
+import cats.effect.{Concurrent, Ref}
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.partialOrder._
+
+import org.tessellation.ext.cats.syntax.next.catsSyntaxNext
+import org.tessellation.schema.currencyMessage._
+import org.tessellation.security.signature.Signed
+
+trait CurrencyMessageStorage[F[_]] {
+  def get(messageType: MessageType): F[Option[Signed[CurrencyMessage]]]
+  def set(message: Signed[CurrencyMessage]): F[Boolean]
+}
+
+object CurrencyMessageStorage {
+  def make[F[_]: Concurrent: Ref.Make]: F[CurrencyMessageStorage[F]] =
+    MapRef.ofSingleImmutableMap[F, MessageType, Signed[CurrencyMessage]](Map.empty).map { mapRef =>
+      new CurrencyMessageStorage[F] {
+        def get(messageType: MessageType): F[Option[Signed[CurrencyMessage]]] =
+          mapRef(messageType).get
+
+        def set(message: Signed[CurrencyMessage]): F[Boolean] =
+          mapRef(message.messageType).modify {
+            case None                                                      => (message.some, true)
+            case Some(m) if message.parentOrdinal === m.parentOrdinal.next => (message.some, true)
```

Comment:
it should be:
```
m.ordinal === message.parentOrdinal
```

where `m.ordinal` is defined as `def ordinal = parentOrdinal.next` using Next instance. That way it's easier to follow what we're actually checking

Reply:
done